server/controllers/SolutionController.js

```diff
@@ -43,13 +45,15 @@ exports.getComments = async (req, res) => {
 
 exports.create = async (req, res) => {
   try {
-    const { challengeId, userId, title, description, diagram } = req.body;
+    const { challengeId, userId, title, description } = req.body;
+    const { file } = req;
```

Comment:
What is this line doing?

Reply:
Removed the diagram from the parsed request body parameters to the file parameter which is handled by the multer middleware. So rather than receiving the diagram in the body of the request that's parsed by the json middleware, we are using multer to automatically save it, and the file parameter contains the information about the file (not the file itself).